src/platform/plugins/shared/controls/public/controls/data_controls/options_list_control/editor_state_manager.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import {
+  StateComparators,
+  initializeStateManager,
+} from '@kbn/presentation-publishing/state_manager';
+import { OptionsListControlState } from '../../../../common/options_list';
+import { DEFAULT_SEARCH_TECHNIQUE } from './constants';
+
+export type EditorState = Pick<
```

Comment:
Do we need a separate type and state manager for the editor? Instead can we use the existing state manager, or at least create one with the same type?

Reply:
See [conversation](https://github.com/elastic/kibana/pull/220688#discussion_r2100734459) about `temporaryStateManager` in  but I am not sure which state manager should be merged with `editorStateManager`.